File: 2019/rp1.php
<?php

function solve($a, $b) {
    $m = ($a + $b) / 2;
    printf("%d\n", $m);
    fscanf(STDIN, "%s", $s);
    if (strcmp($s, "CORRECT") == 0)
        return;
    else if (strcmp($s, "TOO_SMALL") == 0)
        $a = $m + 1;
    else
        $b = $m - 1;
    solve($a, $b);
}

fscanf(STDIN, "%d", $t);
for ($ks = 0; $ks < $t; $ks++) {
    fscanf(STDIN, "%d %d", $a, $b);
    fscanf(STDIN, "%d", $n);
    solve($a + 1, $b);
}
?>

File: 2019/rp1.rb
$stdout.sync = true

def solve(a, b)
  m = (a + b) / 2
  puts m
  $stdout.flush
  s = STDIN.gets.chomp
  if s.eql? "CORRECT"
    return
  elsif s.eql? "TOO_SMALL"
    solve(m + 1, b)
  else
    solve(a, m - 1)
  end
end

t = STDIN.gets.chomp.to_i
ks = 1
while ks <= t
  a, b = STDIN.gets.split.map &:to_i;
  n = STDIN.gets.chomp.to_i
  solve(a + 1, b)
  ks = ks + 1
end
